CraftingRevisions/Patches/GameManager_Awake.cs

```diff
@@ -1,19 +1,99 @@
-﻿using HarmonyLib;
+using HarmonyLib;
 using Il2Cpp;
-using MelonLoader;
+using System.Reflection;
+using UnityEngine;
 
 namespace CraftingRevisions.Patches
 {
-	[HarmonyPatch(typeof(GameManager), "Awake")]
-	internal class GameManager_Awake
+	// load the blueprints when the crafting panel is initialized
+	// (needed to be this late otherwise we have no instance of BlueprintManager)
+	[HarmonyPatch(typeof(Panel_Crafting), nameof(Panel_Crafting.Initialize))]
+	internal class Panel_Crafting_Initialize
 	{
 		private static void Postfix()
 		{
-			try { BlueprintMapper.MapBlueprints(); }
-			catch (System.Exception e)
+			// call TLD LoadAllUserBlueprints
+			InterfaceManager.m_Instance.m_BlueprintManager.LoadAllUserBlueprints();
+		}
+	}
+
+	// patched into postfix BlueprintManager.LoadAllUserBlueprints
+	// (BlueprintManager.RemoveUserBlueprints is not guaranteed to get called)
+	[HarmonyPatch(typeof(Il2CppTLD.Gear.BlueprintManager), nameof(Il2CppTLD.Gear.BlueprintManager.LoadAllUserBlueprints))]
+	internal class BlueprintManager_LoadAllUserBlueprints_Postfix
+	{
+		private static void Postfix(Il2CppTLD.Gear.BlueprintManager __instance)
+		{
+			// check we have a HasSet and it has contents
+			if (BlueprintManager.jsonUserBlueprints != null && BlueprintManager.jsonUserBlueprints.Count > 0)
```

Comment:
This if statement isn't needed. The hash set will never be null, and checking the size is redundant.